packages/models/src/lib/category-config.ts

```diff
@@ -0,0 +1,66 @@
+import { z } from 'zod';
+
+/**
+ *
+ * Define Zod schema for the CategoryConfig type
+ *
+ * @example
+ *
+ * // Example data for the CategoryConfig type
+ * const categoryConfigData = {
+ *   slug: 'performance',
+ *   title: 'Performance Metrics',
+ *   description: 'This category includes performance-related metrics.',
+ *   metrics: [
+ *     { ref: 'eslint#max-lines', weight: 0.5 },
+ *     { ref: 'categories:lhci#performance', weight: 0.8 },
+ *   ],
+ * };
+ *
+ * // Validate the data against the schema
+ * const validationResult = categoryConfigSchema.safeParse(categoryConfigData);
+ *
+ * if (validationResult.success) {
+ *   console.log('Valid category config:', validationResult.data);
+ * } else {
+ *   console.error('Invalid category config:', validationResult.error);
+ * }
+ */
+export const categoryConfigSchema = z.object(
+  {
+    slug: z.string({
+      description: 'Human-readable unique ID',
+    }),
+    title: z.string({
```

Comment:
It would be good to set a maximum length for text fields (in all these schemas). The portal will need to enforce a limit anyway (for security, performance and document size limits). 

- For short texts (`slug`s and `title`s), we could have a maximum of e.g. 128 characters.
- For long texts (`description`s with Markdown), we should have a larger maximum, e.g. 65536 (same as GitHub comments).